examples/knative/go/kn-go-echo/server.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+)
+
+func main() {
+	ctx := context.Background()
+
+	client, err := cloudevents.NewClientHTTP()
+	if err != nil {
+		log.Fatalf("creating HTTP client, %v", err)
+	}
+
+	log.Println("listening on :8080")
+	if err := client.StartReceiver(ctx, receive); err != nil {
+		log.Fatalf("start receiver: %s", err.Error())
+	}
+}
+
+func receive(ctx context.Context, event cloudevents.Event) *cloudevents.Event {
+	fmt.Printf("***cloud event***\n%s", event)
+
+	response := event
+	// Change the type here and update trigger filter in function.yaml in order
+	// to avoid an infinite event loop.
+	response.SetType("forwarding of com.vmware.event.router/event")
```

Comment:
This is not a well defined type based on the CE spec I would argue. 
For this exercise, you could append `.echo` to the incoming event and ignore any event with that suffix in your logic to avoid loops.

Reply:
Appending `.echo` to the incoming event will make the even no longer a cloud event. What I can do is add a cloud event field called `isEcho` and check that field. If the field is `true`, I will return an empty cloudevent, `event.Event{}`.

If `isEcho == false`, then I'll go ahead and forward the vent.

Reply:
I am not following you why appending `.echo` to the event type (string) would break CloudEvents/JSON?

Reply:
> you could append .echo to the incoming event

I think I misunderstood you. I interpreted what you said as, "Append `.echo` to the serialized incoming event". Instead, you meant,

> appending .echo to the event type (string) 

Ok, yes. That is much easier and makes more sense. I can do, 

```go
response.SetType("com.vmware.event.router/event.echo")
```

And instead of using Knative trigger filters to look for this type of event, I can filter using logic in the server.go code.

Is that what you mean?

Reply:
Thx, but please don't make it hardcoded but use append syntax on that string since incoming event type might be different depending on the event.

Reply:
Latest commit is now getting the type with `event.Type()`.